shopfloor/services/location_content_transfer.py

```diff
@@ -448,23 +457,79 @@ def _find_transfer_move_lines(self, location):
-        self._write_destination_on_lines(move_lines, dest_location)
+    def _set_all_destination_lines_and_done(
+        self, pickings, move_lines, dest_location, package=None
+    ):
+        if package and not package.location_id:
+            # Using an empty package
+            package.location_id = dest_location
+        self._write_destination_on_lines(move_lines, dest_location, package)
         stock = self._actions_for("stock")
         stock.validate_moves(move_lines.move_id)
 
     def _lock_lines(self, lines):
         """Lock move lines"""
         self._actions_for("lock").for_update(lines)
 
-    def set_destination_all(self, location_id, barcode, confirmation=False):
+    def _is_package_empty(self, package):
+        return not bool(package.quant_ids)
+
+    def _set_dest_get_empty_pack(self, package_id):
+        empty_package = self.env["stock.quant.package"]
+        if package_id:
+            empty_package = self.env["stock.quant.package"].browse(package_id).exists()
+        return empty_package
+
+    def _set_dest_handle_barcode(self, barcode):
+        message = None
+        scanned_package = self.env["stock.quant.package"]
+        scanned_location = self._actions_for("search").location_from_scan(barcode)
+        if not scanned_location:
+            scanned_package = self._actions_for("search").package_from_scan(barcode)
+        if not scanned_location and not scanned_package:
+            message = self.msg_store.barcode_not_found()
+        return scanned_location, scanned_package, message
+
+    def _set_dest_validate_location(self, move_ids, location, package, empty_package):
+        # Used by the module `shopfloor_location_content_transfer_force_package`
+        message = None
+        destination_location = location or package.location_id
+        if destination_location:
+            if not self.is_dest_location_valid(move_ids, destination_location):
+                message = self.msg_store.dest_location_not_allowed()
+        return message
+
+    def _set_dest_handle_empty_package(self, scanned_package, empty_package):
+        """Manage the empty package on set_destination... calls.
+
+        If the user scanned an empty package, return a response to the same
+        endpoint to scan a location.
+        Else if the user previously scanned an empty package (passed in the
+        request data) it will be used to move the goods.
+        """
+        message = None
+        if scanned_package and self._is_package_empty(scanned_package):
+            # New empty pack scanned, ask for a destination location to use.
+            empty_package = scanned_package
+            message = self.msg_store.package_selected_is_empty(scanned_package)
+        elif empty_package:
+            if not self._is_package_empty(empty_package):
+                message = self.package_not_empty_anymore(empty_package)
+                empty_package = self.env["stock.quant.pacakge"]
```

Comment:
Split in 2 different handler. One is when scanning a package, one is when scanning a location.

If the package is not empty but already on the scanned location, it is not an issue.

Reply:
I thought it would be good to inform the user.
Next step for the user if he wants to use this package is to scan it again.